optuna/samplers/_nsga2/sampler.py

```diff
@@ -154,18 +177,35 @@ def sample_relative(
         generation = parent_generation + 1
         study._storage.set_trial_system_attr(trial_id, _GENERATION_KEY, generation)
 
+        dominates_func = _dominates if self._constraints_func is None else _constrained_dominates
         if parent_generation >= 0:
-            p0 = self._select_parent(study, parent_population)
+            # We choose a child based on the specified crossover method.
             if self._rng.rand() < self._crossover_prob:
-                p1 = self._select_parent(
-                    study, [t for t in parent_population if t._trial_id != p0._trial_id]
+                child = crossover(
+                    self._crossover,
+                    study,
+                    parent_population,
+                    search_space,
+                    self._rng,
+                    self._swapping_prob,
+                    dominates_func,
                 )
             else:
-                p1 = p0
+                parent_population_size = len(parent_population)
+                child = parent_population[self._rng.choice(parent_population_size)].params
 
-            study._storage.set_trial_system_attr(
-                trial_id, _PARENTS_KEY, [p0._trial_id, p1._trial_id]
-            )
+            params_len = len(child)
+            if self._mutation_prob is None:
+                mutation_prob = 1.0 / max(1.0, params_len)
+            else:
+                mutation_prob = self._mutation_prob
+
+            params = {}
+            for param_name in child.keys():
+                if self._rng.rand() >= mutation_prob:
+                    params[param_name] = child[param_name]
+
+            return params
```

Comment:
Let me suggest the following change to improve code readability.
But, this is a matter of personal preference. So feel free to reject this change if you like the current code.
```suggestion
            for param_name in child.keys():
                if self._rng.rand() < mutation_prob:
                    # Do mutation.
                    # The value of this key will be newly sampled in the `sample_independent()` method.
                    del child[param_name]

            return child
```